packages/web-components/fast-foundation/src/carousel/carousel.ts

```diff
@@ -0,0 +1,561 @@
+import { attr, observable, DOM } from "@microsoft/fast-element";
+import { Tabs } from "../tabs";
+import {
+    keyCodeSpace,
+    keyCodeEnter,
+    wrapInBounds,
+    limit,
+    KeyCodes,
+} from "@microsoft/fast-web-utilities";
+import { applyMixins } from "../utilities";
+import { ARIAGlobalStatesAndProperties } from "../patterns";
+
+/**
+ * The pattern type that the Carousel follows.
+ * See {@link https://w3c.github.io/aria-practices/#wai-aria-roles-states-and-properties-4 | ARIA Carousel Patterns
+ */
+export enum CarouselPattern {
+    basic = "basic",
+    tabbed = "tabbed",
+}
+
+/**
+ * An Carousel Custom HTML Element.
+ * Implements the {@link https://www.w3.org/TR/wai-aria-1.1/#carousel | ARIA carousel }.
+ *
+ * @public
+ */
+export class Carousel extends Tabs {
+    /**
+     * Determines if the element should start rotation or not on page load.
+     * @public
+     * @defaultValue - true
+     * @remarks
+     * HTML Attribute: autoplay
+     */
+    @attr({ mode: "fromView" })
+    public autoplay: boolean = true;
+
+    /**
+     * Determines if the element should loop slides or not.
+     * @public
+     * @defaultValue - true
+     * @remarks
+     * HTML Attribute: loop
+     */
+    @attr({ mode: "fromView" })
+    public loop: boolean = true;
```

Comment:
Just catching this, I don't believe we can set attributes like this, you have to wait for the `connectedCallback` to initialize them. See https://github.com/microsoft/fast/pull/3494 as an example